internal/runners/packages/searchView.go

```diff
@@ -0,0 +1,134 @@
+package packages
+
+import (
+	"fmt"
+	"os"
+	"strconv"
+	"strings"
+
+	"github.com/ActiveState/cli/internal/errs"
+	"github.com/ActiveState/cli/internal/locale"
+	"github.com/charmbracelet/bubbles/viewport"
+	tea "github.com/charmbracelet/bubbletea"
+	"github.com/charmbracelet/lipgloss"
+	"golang.org/x/term"
+)
+
+const (
+	verticalMargin = 7
+)
+
+type errMsg error
+
+type view struct {
+	width        int
+	height       int
+	content      string
+	remaining    int
+	packageNames []string
+	ready        bool
+	err          error
+	viewport     viewport.Model
+}
+
+func NewView() (*view, error) {
+	width, height, err := term.GetSize(int(os.Stdout.Fd()))
+	if err != nil {
+		return nil, errs.Wrap(err, "Could not get terminal size")
+	}
+
+	return &view{
+		width:  width,
+		height: height,
+	}, nil
+}
+
+func (v *view) Init() tea.Cmd {
+	return nil
+}
+
+func (v *view) Update(msg tea.Msg) (tea.Model, tea.Cmd) {
+	var (
+		cmd  tea.Cmd
+		cmds []tea.Cmd
+	)
+
+	switch msg := msg.(type) {
+	case tea.KeyMsg:
+		stringMsg := strings.ToLower(msg.String())
+		switch stringMsg {
+		case "q", "ctrl+c":
+			return v, tea.Quit
+		case "up":
+			lines := v.viewport.LineUp(1)
+			for _, l := range lines {
+				if strings.Contains(l, "Name") {
+					v.remaining++
+				}
+			}
+			return v, nil
+		case "down":
+			lines := v.viewport.LineDown(1)
+			for _, l := range lines {
+				if strings.Contains(l, "Name") {
+					if v.remaining <= 0 {
+						v.remaining = 0
+					} else {
+						v.remaining--
+					}
+				}
+			}
+			return v, nil
+		}
```

Comment:
It seems viewport also supports page up and page down, which this does not check for. Once you start using an alternate mode of scrolling not covered here the remaining number is out of sync. Wonder if there is a scroll event we can listen for?

Reply:
There does not appear to be a scroll event. The viewport has a scroll percent function that will return the remaining percentage left. We might want to consider that if we're going to implement alternate scrolling methods. We could also look to some sort of index calculation. The mock just had `down` to scroll so that's all that is here. Do we want to take this further or file a follow up?

Reply:
We should solve it as part of this ticket. I think showing a percentage indication would be fine too, it's just about telling the user that there is something to scroll to since there is no scrollbar.

Reply:
Managed to get this and mouse scroll working with an index.